File: CPP/Program/Karakter.cpp
#pragma once
#include <iostream>
#include <string>
#include "Status.cpp"
using namespace std;

class Karakter {
protected:
    string Nama;         // Nama karakter
    int TinggiBadan;     // Tinggi badan karakter
    int Umur;            // Umur karakter
    Status status;       // Status karakter

public:
    // Konstruktor
    Karakter(string nama, int tinggi, int umur, Status s)
        : Nama(nama), TinggiBadan(tinggi), Umur(umur), status(s) {}

    // Setter
    void setNama(string n) { Nama = n; }
    void setTinggiBadan(int t) { TinggiBadan = t; }
    void setUmur(int u) { Umur = u; }
    void setStatus(Status s) { status = s; }

    // Getter
    string getNama() { return Nama; }
    int getTinggiBadan() { return TinggiBadan; }
    int getUmur() { return Umur; }
    Status getStatus() { return status; }

    // Menampilkan info karakter
    virtual void tampilkanInfo() {
        cout << "Nama: " << Nama
             << ", Tinggi: " << TinggiBadan
             << ", Umur: " << Umur << endl;
        status.tampilkanStatus();
    }
};

File: Java/Program/Karakter.java
public class Karakter {
    // Atribut dasar karakter
    private String nama;
    private double tinggiBadan;
    private int umur;

    // Konstruktor
    public Karakter(String nama, double tinggiBadan, int umur) {
        this.nama = nama;
        this.tinggiBadan = tinggiBadan;
        this.umur = umur;
    }

    // Getter dan setter nama
    public String getNama() { return nama; }
    public void setNama(String nama) { this.nama = nama; }

    // Getter dan setter tinggi badan
    public double getTinggiBadan() { return tinggiBadan; }
    public void setTinggiBadan(double tinggiBadan) { this.tinggiBadan = tinggiBadan; }

    // Getter dan setter umur
    public int getUmur() { return umur; }
    public void setUmur(int umur) { this.umur = umur; }

    // Menampilkan info karakter
    public void tampilkanInfo() {
        System.out.println("Nama: " + nama);
        System.out.println("Tinggi Badan: " + tinggiBadan);
        System.out.println("Umur: " + umur);
    }
}
